storm-client/src/jvm/org/apache/storm/grouping/ShuffleGrouping.java

```diff
@@ -26,38 +26,25 @@
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
-import java.util.concurrent.atomic.AtomicInteger;
+
 
 public class ShuffleGrouping implements CustomStreamGrouping, Serializable {
-    private Random random;
     private ArrayList<List<Integer>> choices;
-    private AtomicInteger current;
+    private int current = 0 ;
```

Comment:
You changed this to non threadsafe. Is this instance not shared by multiple components?